src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -89,6 +91,8 @@ public function getUser()
         if (null === $this->tokenStorage) {
             if (null === $this->container) {
                 throw new \RuntimeException('The "app.user" variable is not available.');
+            } elseif (!$this->container->has('security.context')) {
+                return;
```

Comment:
What about deprecating that behaviour with Symfony 2.8 and throw the exception starting with 3.0?